internal/use_cases/create_pull_request.go

```diff
@@ -43,8 +48,50 @@ type CreatePullRequest struct {
 	BranchProvider          domain.BranchProvider
 }
 
+func normalizeTemplatePath(git domain.GitProvider, path string) (string, error) {
```

Comment:
suggestion: instead of passing the provider, create this function as a struct method for `CreatePullRequest` and use the git provider within the struct for this function.